javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistTypeDeclarationAdapter.java

```diff
@@ -110,7 +112,9 @@ private void collectDeclaredFields(CtClass ctClass, List<ResolvedFieldDeclaratio
      */
     public List<ResolvedReferenceType> getAncestors() {
         return Collections.singletonList(
-                JavassistFactory.typeUsageFor(ctClass, typeSolver).asReferenceType()
+                JavaParserFacade.get(typeSolver)
+                        .classToResolvedType(Annotation.class)
+                        .asReferenceType()
         );
     }
```

Comment:
Potentially silly question -- for what reason are we able to _always_ return `Annotation` here? 

If we were within `JavassistAnnotationDeclaration.java` I could understand (the superclass of an annotation is always `Annotation`), but this is within the more general `JavassistTypeDeclarationAdapter.java`. 

What have I missed? :)

Reply:
🤦 You are right! Thanks for pointing it out! For some reason i assumed it was an adapter only for AnnotationDeclaration but it's an adapter for all the TypeDeclaration. I'll make it get all the classes from the `CtClass` :)